chain/messagepool/messagepool.go

```diff
@@ -474,6 +474,15 @@ func (mp *MessagePool) TryForEachPendingMessage(f func(cid.Cid) error) error {
 }
 
 func (mp *MessagePool) resolveToKey(ctx context.Context, addr address.Address) (address.Address, error) {
+	//if addr is not an ID addr, then it is already resolved to a key
+	if addr.Protocol() != address.ID {
+		return addr, nil
+	}
+	return mp.resolveToKeyFromID(ctx, addr)
+}
+
+func (mp *MessagePool) resolveToKeyFromID(ctx context.Context, addr address.Address) (address.Address, error) {
```

Comment:
@Stebalien In what cases can an actor have both an f4 and an f2 address? Is that just EVM actors (not including EthAccounts)?

I wanna make sure you can never have an actor have both an f4 and an f2 address AND serving as a top-level sender, because that would make things funky here.

(Not really related to changes in this PR)

Reply:
Just EVM actors, at least for now. Account abstraction may someday change this, but that's a whole can of worms.